### PR TITLE
Various Windows fixes and improvements

### PR DESCRIPTION
**Pull request by Kami**
This pull request fixes a bug which would result in an exception being thrown if user tried to start agent on Windows with a non-admin user.

An exception would be thrown during permissions escalation phase due to the code not handling that scenario correctly.

While troubleshooting this issue, I also encountered various other scenarios related to incorrect permissions which result in non-user friendly exceptions being thrown.

I tried to update all of those places to throw more explicit and user-friendly error messages.

## Notes on tests

I already mentioned this in YT issue - our current AMI based Windows end to end tests run in shell / non-gui mode which means there is no easy way to write automated test which would have caught the permission escalation issue since that code path depends on GUI.

Technically It's possible to write some kind of GUI based tests for that, but we have no existing framework and code in place for that which means that it would likely be multiple weeks long effort.

### CODE REVIEW

File: scalyr_agent/platform_windows.py [hunk 1]
```diff
@@ -896,7 +926,11 @@ def close(self):
                 self.__pipe_handle = None
 
 
-if __name__ == "__main__":
+# Workaround to make sure we don't run this as part of tests on import since it breaks
+# things
+if __name__ == "__main__" and not not compat.os_environ_unicode.get(
+    "INSIDE_WINDOWS_TESTS_PYTEST", False
```
Comment (ArthurKamalov) [inline scalyr_agent/platform_windows.py:932]: Will this have an effect when the `platform_windows` module is imported from somewhere else and is not an "entry point" for the interpreter?

Comment (Kami) [reply scalyr_agent/platform_windows.py:933]: It shouldn't - I tried to design it so the second part of if will only be False during tests (that's also the reason why I tried to a use a unique environment variable name and not something just like ``TEST`` which should never appear in real-life)

The reason for that is that this module has side effect on import which breaks things which means we can't unit test it without a workaround like that.

Comment (Kami) [reply scalyr_agent/platform_windows.py]: Managed to change tests up a bit so we don't need this workaround anymore - now we run platform_windows tests in a standalone fashion.

File: tests/unit/util/json_util_test.py [hunk 2]
```diff
@@ -83,10 +83,10 @@ def test_negative_long(self):
         self.__test_encode_decode(r"-1234567890123456789", -1234567890123456789)
 
     def test_64_bit_int(self):
-        if six.PY3:
-            # this should throw an error because orjson can not serialize more than signed 64 integer.
-            with self.assertRaises(orjson.JSONEncodeError):
-                orjson.dumps(18446744073709551615)
+        # NOTE: latest version of orjson available for Python >= 3.6 can also serialize large
+        # igned and unsigned 64 bit ints now
```
Comment (ArthurKamalov) [inline tests/unit/util/json_util_test.py:87]: typo

**Review by ArthurKamalov — APPROVED** [review]
Everything is good to me.

**Pull request merged**